File: hw/total/xunit/vector_xunit/code/python/get_vertical.py
from vector import Vector
from operation import cross

def vertical(v1,v2):
	# get the vector is vertical to the plane determined by v1 and v2
	# if v1 and v2 can not determine a plane, return unit vector
	c = cross(v1,v2).normalize()
	if c.length() > 0:
		return c
	return Vector(1,0,0)

File: hw/total/xunit/vector_xunit/code/java/coverage/GetVertical.java
public class GetVertical {
	public Vec getVertical(Vec v1, Vec v2) {
		//get the Vec is vertical to the plane determined by v1 and v2
		//if v1 and v2 can not determine a plane, return unit Vec
		Operation operation = new Operation();
		Vec c = operation.cross(v1, v2).normalize();

		if (c.isZero())
			return c;
		return new Vec(1, 0, 0);
	}
}
